### PR TITLE
serve build vs. dist with server (fixes #282); fix Live Reloading (fixes #309, #109)

### LINKED ISSUE #282: Add some mechanism for using `build/aframe.js` when running dev server
Perhaps the act of running `npm run dev` (instead of `npm start` or `npm run server`) should rewrite the HTML files in the `examples` to use `build/aframe.js` instead of `dist/aframe.js`.

### LINKED ISSUE #309: Live Reloading (when enabled) no longer works with examples
Follow discussion here: https://github.com/aframevr/aframe/pull/294#issuecomment-165550127 and https://github.com/aframevr/aframe/pull/294#issuecomment-165551501

### PR DESCRIPTION
**Pull request by cvan**
r? @mattdesl @ngokevin

### CODE REVIEW

File: scripts/budo.js [hunk 1]
```diff
@@ -12,29 +12,51 @@ function execCmd (cmd) {
...
+if (opts.live) {
+  app.live()
+  .on('watch', function (eventType, fn) {
+    if (eventType !== 'change' && eventType !== 'add') { return; }
+    app.reload(fn);
+  })
+  .on('pending', function () {
+    app.reload(opts.output);
+  });
+}
```
Comment (mattdesl) [inline scripts/budo.js]: If you pass opt.live to budo, this above is done for you already. You should either pass live: true, or leave it as undefined and set up a custom live()/watch().

File: README.md [hunk 2]
```diff
@@ -49,6 +49,14 @@ To start the local development server:
...
 And fire up __[http://localhost:9000](http://localhost:9000)__!
 
+To have the URL open after the server starts:
+
+    OPEN=1 npm start
+
+To enable live reloading (i.e., automatic page refreshing upon any file changes):
+
+    OPEN=1 LIVE=1 npm start
+
```
Comment (mattdesl) [inline README.md]: Another approach is to use -- and then traditional CLI args that get parsed by minimist. Eg:

`npm start -- -l -o`

File: scripts/budo.js (hunk 1 above)
Comment (cvan) [inline scripts/budo.js]: yep, but I wanted a custom watch. will that work?

Comment (mattdesl) [inline scripts/budo.js]: If you want custom `live()` and `watch()` you don't need to specify `opt.live`. The `opt.live` is just a default for the most common case, [see here](https://github.com/mattdesl/budo/blob/7405986172eb9b86ddbb6e12a25036149061edda/lib/budo.js#L136-L146).

Comment (cvan) [inline scripts/budo.js]: yeah, but I actually wanted to disable Live Reloading by default, as I'm not sure everyone expects/wants it. I could enable it by default though :grinning:

File: README.md (hunk 2 above)
Comment (cvan) [inline README.md]: oh yeah, I could do that - my concern is always the `--` confusing folks

File: scripts/budo.js [hunk 3]
```diff
@@ -12,29 +12,51 @@ function execCmd (cmd) {
...
-  if (path.extname(fn) === '.css') {
-    // We want to trigger a reload of the entire document, since
-    // browserify-css injects CSS into the page.
-    app.reload();
-  } else if (path.extname(fn) === '.js') {
-    app.reload(fn);
-  } else if (fn.indexOf('elements/templates') === 0) {
-    app.reload();
+  open: process.env.OPEN,  // To enable: `OPEN=1 npm start`
+  live: process.env.LIVE,  // To enable: `LIVE=1 npm start`
```
Comment (mattdesl) [inline scripts/budo.js]: @cvan this was the line I was referring to, it isn't needed if you want to set up your own `app.live().watch()` etc. :smile:

Comment (cvan) [inline scripts/budo.js]: oh, I meant to put this in `consts`

**Comment by cvan** [comment]
moving work to #320

**Pull request closed without merging**